Automattic/wp-calypso
# Checkout: Use rearranged radio button variant picker

#### Proposed Changes

This PR replaces the billing term variant picker in checkout for WordPress.com with a set of radio buttons instead of a dropdown menu. **Checkout for Jetpack sites will remain unchanged** and continue to use the dropdown menu. The radio buttons implement the variant picker design laid out in https://github.com/Automattic/wp-calypso/issues/65968#issuecomment-1324384020. This follows a series of adjustments made to the variant picker to improve its usefulness (see pbOQVh-24K-p2).

This new UI is only made available for people in the Explat AB test experiment group `calypso_checkout_variant_picker_radio_2212`. A p2 post for this experiment is here: pbxNRc-2ce-p2

#### Screenshots

Before this change:

<img width="567" alt="Screen Shot 2022-11-22 at 8 10 22 PM" src="https://user-images.githubusercontent.com/2036909/203451532-a6cea1a0-5dc0-4889-9fc1-a2db1c3a380f.png">

<img width="570" alt="Screen Shot 2022-11-22 at 8 10 31 PM" src="https://user-images.githubusercontent.com/2036909/203451538-63bdb0ff-d4e2-42c1-bc99-f18da8aca7f6.png">

After this change, a monthly plan in the cart:

<img width="569" alt="Screen Shot 2022-11-22 at 8 05 05 PM" src="https://user-images.githubusercontent.com/2036909/203451147-7c311190-d47c-48fd-872f-02918a5c62c9.png">

After this change, an annual plan in the cart:

<img width="567" alt="Screenshot 2022-12-01 at 8 29 49 PM" src="https://user-images.githubusercontent.com/2036909/205194279-557b5f0e-55d7-4c74-8acc-7bc8f60c86ea.png">

After this change, an annual plan and a first-year discount in the cart:

<img width="568" alt="Screen Shot 2022-11-22 at 8 06 14 PM" src="https://user-images.githubusercontent.com/2036909/203451157-e94a77a8-2906-4438-8431-f4dc37adae76.png">

#### Testing Instructions

There are unit tests which cover the new UI, but you can test it manually by assigning yourself to the `treatment` experiment group in Abacus: 20943-explat-experiment

Add various products to your cart and visit checkout to see how they look. Specifically, WordPress.com plans will trigger the radio button variant picker. Jetpack plans will trigger the dropdown variant picker, but that should be unchanged.

Click to select different plans in the radio button variant picker and verify that they change as expected.
